From 8361a9b8cb6a9c71b7cf884a87b2532d8367c185 Mon Sep 17 00:00:00 2001
From: Brian Norris <computersforpeace@gmail.com>
Date: Fri, 30 Oct 2015 20:33:30 -0700
Subject: [PATCH 11/11] mtd: physmap_of: assign parent for the concatenated MTD

If there is more than one map region for this device, then the
concatenated MTD will not have a parent device assigned to it -- only
the sub-devices (which are not actually registered with the framework)
will have their parents assigned. Let's assign the concatenated device
correctly.

Signed-off-by: Brian Norris <computersforpeace@gmail.com>
Reviewed-by: Boris Brezillon <boris.brezillon@free-electrons.com>
---
 drivers/mtd/maps/physmap_of.c | 1 +
 1 file changed, 1 insertion(+)

--- a/drivers/mtd/maps/physmap_of.c
+++ b/drivers/mtd/maps/physmap_of.c
@@ -271,6 +271,7 @@ static int of_flash_probe(struct platfor
 	if (err)
 		goto err_out;
 
+	info->cmtd->dev.parent = &dev->dev;
 	mtd_set_of_node(info->cmtd, dp);
 	mtd_device_parse_register(info->cmtd, part_probe_types_def, NULL,
 			NULL, 0);
